src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/SourceWriter.cs

```diff
@@ -581,6 +586,7 @@ private void WriteGreenFactory(Node nd, bool withSyntaxFactoryContext = false)
             if (nd.Name != "SkippedTokensTriviaSyntax" &&
                 nd.Name != "DocumentationCommentTriviaSyntax" &&
                 nd.Name != "IncompleteMemberSyntax" &&
+                nd.Name != "AttributeSyntax" &&
```

Comment:
💭 I wasn't sure why this case was needed. It seems like the flags will always be correct, whether or not this is cached.

Reply:
➡️ Answer was given in a different comment https://github.com/dotnet/roslyn/pull/72070/files#r1488427267

Reply:
Note: i looked into potentially being able to cache thse nodes, and it quickly became quite complex.  The core issue is that the caching logic really depends today on only caching things *without* flag bits on them (except for IsNotMissing bit).  Trying to flow through that it's also ok to have other bits was not easy.  And that tracking has to happen at the cached node level itself, as well as when combining the info from the child nodes being combined.  

overall, it seems fien that we don't have perfect caching here for attribute nodes.